modules/password/password.go

```diff
@@ -0,0 +1,58 @@
+package password
+
+import (
+	"math/rand"
+	"regexp"
+	"sync"
+	"time"
+
+	"code.gitea.io/gitea/modules/setting"
+)
+
+var matchComplexities = map[string]regexp.Regexp{}
+var matchComplexityOnce sync.Once
+
+// CheckPasswordComplexity return True if password is Complexity
+func CheckPasswordComplexity(pwd string) bool {
+	if len(setting.PasswordComplexity) > 0 {
+		matchComplexityOnce.Do(func() {
+			for key, val := range setting.PasswordComplexity {
+				matchComplexity := regexp.MustCompile(val)
```

Comment:
Assuming, `val` is an invalid regex, I believe this will cause a panic. Is there a way to make this fail on boot/start up?  Instead of a panic at runtime?

Reply:
@adelowo These are hardcoded regexps. They will not fail. See:

https://github.com/T-M-A/gitea/blob/59f2a89c6e9d7f72b6a9138ae12fcc1d311d16d8/modules/setting/setting.go#L778-L783

Reply:
Ha I see.

Reply:
Few months ago, the PR allowed the user to set regex from the config file I think..

Thanks @guillep2k

Reply:
I could not find a variant of one regular expression satisfying all conditions at once. 👎

Reply:
> Few months ago, the PR allowed the user to set regex from the config file I think..


I could not find a variant of one regular expression satisfying all conditions at once. I had to redo it on a harsh condition